x-pack/plugins/infra/server/lib/alerting/metric_threshold/lib/evaluate_alert.ts

```diff
@@ -121,7 +122,10 @@ const getMetric: (
   const intervalAsSeconds = getIntervalInSeconds(interval);
   const intervalAsMS = intervalAsSeconds * 1000;
 
-  const to = roundTimestamp(timeframe ? timeframe.end : Date.now(), timeUnit);
+  const to = moment(timeframe ? timeframe.end : Date.now())
+    .add(1, timeUnit)
+    .startOf(timeUnit)
+    .valueOf();
   // We need enough data for 5 buckets worth of data. We also need
   // to convert the intervalAsSeconds to milliseconds.
   const minimumFrom = to - intervalAsMS * MINIMUM_BUCKETS;
```

Comment:
Add a `TODO` here to move this to the rate aggregation

Reply:
👍